tox.ini

```diff
@@ -10,7 +10,7 @@ deps =
     -r{toxinidir}/requirements.txt
     PyGObject
     pytest-cov
-    pytest==3.5.1
+    pytest>=3.5.1
```

Comment:
Why is pytest so special that we need to pin its version? We do not pin most of the other versions including the package in the docker containers.

Reply:
It is not, we should have pin the versions in the CI env in order to avoid cases where a new released version takes down the CI jobs. (exactly what happened just now, and I seen happening a lot in other projects)
It just needs to be controlled and incremented only when confirmed to work.

The docker is pined by definition, it only may change when creating a new image, but that is controlled and verified.

Reply:
The problem here is that we pinned a version and nobody is monitoring for new releases IMHO (which is also an annoying job). I would say that it is a lot easier to just revert to a previous version in case there is some incompatibility with a future version (or even better do adopt) than to realize after a long time that all the pinned versions are ancient and a lot of necessary changes  accumulated.

Reply:
We pinned one component and not the other which depends on it. I see no direct reason to use the latest testing tools, it is a matter of stability. We already had incidents when tests failed due to a bug introduced in the tools.

I do not mind using the latest version, but it must be controlled.
We do not merge patches that are not tested, I expect to follow the same rule with the testing setup dependencies.

Reply:
BTW, removing the pinning will cause all to use the latest, probably solving the problem as well. But again, I strongly suggest following experience and pinning the testing env to control it.

Reply:
Please discuss this in another PR or issue.
For me, I hate hard coded fix version dependency.

Reply:
@EdDev Please don't push commit direct to upstream without PR review.

Reply:
> @EdDev Please don't push commit direct to upstream without PR review.

What are you talking about? No one did that as far as I know.

Reply:
@cathay4t I reviewed the change and accepted it since it mainly changed the version.